helpers/core.js

```diff
@@ -198,10 +199,20 @@ var helpers = {
 	},
 	'with': function (expr, options) {
 		var ctx = expr;
-		expr = resolve(expr);
-		if ( !! expr) {
-			return options.fn(ctx);
+		if(!options) {
+			// hash-only case if no current context expression
+			options = expr;
+			expr = true;
+			ctx = options.hash;
+		} else {
+			expr = resolve(expr);
+			if(options.hash && !isEmptyObject(options.hash)) {
+				// presumably rare case of both a context object AND hash keys
+				// Leaving it undocumented for now, but no reason not to support it.
+				ctx = options.scope.add(options.hash).add(ctx);
```

Comment:
Do you think it would be useful to add the hash last? I would think someone might want to use the hash to overwrite something from the context, like

```
"{{#with person first=person.nickname}}"
```

Reply:
Tough call.  If you do that, then the original `ctx` there won't be the context object, which is what we'd expect for e.g. `{{#with foo bar=baz}}{{./quux}}{{/with}}`, and I can't really think of any _good_ way to muck with the precedence otherwise.